### PR TITLE
feat: added ESS computation for aldp and config used in camera-ready version of the paper

### PR DESCRIPTION
**Pull request by VincentStimper**
* Added a script that can be used to compute the forward and reverse ESS of the trained models for alanine dipeptide on the test set
* Added the config files for the alanine dipeptide runs that were included in the camera-ready version of the paper accepted at NeurIPS 2023
* Some minor fixes to make the aldp training script run for the config

### CODE REVIEW

File: eacf/train/max_lik_train_and_eval.py [hunk 1]
```diff
@@ -172,20 +172,24 @@ def get_eval_on_test_batch_with_further(
...
     return log_w_joint, info
 
 
 def eval_non_batched(params: AugmentedFlowParams, single_feature: chex.Array,
                      key: chex.PRNGKey, flow: AugmentedFlow,
                      n_samples: int,
                      inner_batch_size: int,
-                     target_log_prob: Callable = None):
+                     target_log_prob: Callable = None,
+                     target_log_prob_traceable: bool = True):
```
Comment (lollcat) [inline eacf/train/max_lik_train_and_eval.py:187]: Why is this necessary? It seems like in the case that `target_log_prob` is not traceable that it should not be passed to this function.

File: examples/aldp_eval/eval_test.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    eval_on_test_batch_fn = partial(get_eval_on_test_batch_with_further,
+                                    flow=flow, K=K, test_invariances=True,
+                                    target_log_prob=None)
+    eval_batch_free_fn = partial(eval_non_batched,
+                                 single_feature=test_data.features[0],
+                                 flow=flow,
+                                 n_samples=n_points,
+                                 inner_batch_size=batch_size,
+                                 target_log_prob=target_log_p_x_fn,
+                                 target_log_prob_traceable=False)
```
Comment (lollcat) [inline examples/aldp_eval/eval_test.py:91]: Am I right that `target_log_p_x_fn` will not be used if `target_log_prob_traceable=False`. In this case it seems simpler to just not pass `target_log_p_x_fn` to this function in the first place rather than needing the additional flag.

**Review by lollcat — APPROVED** [review]
LGTM! Left one comment re the `target_log_prob_traceable` flag which I think should be removed.

File: eacf/train/max_lik_train_and_eval.py (hunk 1 above)
Comment (VincentStimper) [reply eacf/train/max_lik_train_and_eval.py:187]: There is a fine difference: if `traget_lob_prob_tracable = False` but the `target_lob_prob` is still passed, the log prob is not computed within the `forward` method used in the `jax.lax.scan` call, since this would require the `target_lob_prob` to be tracable, but it is computed outside of it just before the ESS is computed, which we want to get.
Hence, I would leave it as it is until we get a better implementation of the aldp distribution.

File: examples/aldp_eval/eval_test.py (hunk 2 above)
Comment (VincentStimper) [reply examples/aldp_eval/eval_test.py:91]: See [this comment](https://github.com/lollcat/se3-augmented-coupling-flows/pull/67#discussion_r1348641471).
